api/src/libs/healthCheck.ts

```diff
@@ -0,0 +1,46 @@
+import { LogWriterType } from "@libs/logWriter";
+import axios, { AxiosError, AxiosResponse } from "axios";
+
+type Status = "PASS" | "FAIL" | "ERROR";
+type StatusResult = Record<string, Status>;
```

Comment:
[sand/boulder] I don't love the result of these being string operations. Ideally you'd have one of these two different return types:

- 0 (C's `EXIT_SUCCESS`) for success, non-0 (C's `EXIT_FAILURE`) for failure, so that your exit code can just be the sum of the exit codes for your operation.
- the HTTP status of the operation.

The first makes more sense to me, as the result of your lambda shouldn't be a string, object, or array but rather an exit code. If you wanted to return anything from the lambda, it would in theory be the resultant objects from each of the health checks as a structured object. But you'd still set the exit code.

Can you tell me more about your approach here and why a `Record<string, Status>` is the ideal return from the lambda?

Reply:
[sand] Also, opportunity for refactor: I don't love the names `Status` and `StatusResult` here.